builder/prepare-ubuntu.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+#
+# This shell script is meant to prepare a building/exec environment for libbpfgo.
+#
+
+die() {
+    echo "ERROR: ${*}"
+    exit 1
+}
+
+info() {
+    echo "INFO: ${*}"
+}
+
+USER="${1}"
+[ -z "${USER}" ] && USER="root"
+USERHOME=$(eval echo ~"${USER}")
+[ -d "${USERHOME}" ] || die "directory ${USERHOME} doesn't exist"
+
+[ -z "${GOVERSION}" ] && GOVERSION="1.17.13"
+[ -z "${ARCH}" ] && ARCH="amd64"
+
+CMDS="wget rm tar chown apt-get"
+for cmd in ${CMDS}; do
+    command -v "${cmd}" >/dev/null 2>&1 || die "missing required tool ${cmd}"
+done
+
+install_go() {
+    info "Installing Go ${GOVERSION}"
+
+    local file="go${GOVERSION}.linux-${ARCH}.tar.gz"
+    local link="https://golang.org/dl/${file}"
+    sudo wget --progress=bar:force "${link}" -P /tmp || die "could not download ${link}"
+
+    local dest="/usr/local"
+    sudo rm -rf "${dest}/go"
+    sudo tar -C "${dest}" -xzf "/tmp/${file}" || die "could not untar ${file} in ${dest}"
+    sudo chown -R root:root "${dest}/go" || die "could not chown"
+
+    local home_profile="${USERHOME}/.profile"
+    echo "export PATH=${PATH}:${dest}/go/bin" | sudo tee -a "${home_profile}"
+
+    info "Go ${GOVERSION} installed"
+}
+
+install_pkgs() {
+    # silence 'dpkg-preconfigure: unable to re-open stdin: No such file or directory'
+    export DEBIAN_FRONTEND=noninteractive
+
+    sudo -E apt-get update || die "coud not update package list"
+    for pkg in "${@}"
+    do
+      info "Installing ${pkg}"
+      sudo -E apt-get install -y "${pkg}" || die "could not install ${pkg}"
+      info "${pkg} installed"
+    done
+}
+
+# startup
+
+info "Starting preparation"
+
+install_go
+
+install_pkgs \
+    coreutils bsdutils findutils \
+    build-essential pkgconf \
+    llvm-12 clang-12 \
```

Comment:
See ? Here you're using llvm-XX from the OS (and it is the correct thing since Ubuntu provides many toolchains in its default repository, I would do the same with go).